packages/sign-client/src/controllers/engine.ts

```diff
@@ -1336,7 +1360,7 @@ export class Engine extends IEngine {
       if (!request) continue;
 
       try {
-        this.processRequest(request);
+        await this.processRequest(request);
         // small delay to allow for any async tasks to complete
         await new Promise((resolve) => setTimeout(resolve, 300));
```

Comment:
Is the timeout still needed ? You're awaiting `processRequest` so it shouldn't be ?
```suggestion
        await this.processRequest(request);
```